src/objectprotocol/number.rs

```diff
@@ -0,0 +1,323 @@
+    ///
+    /// Invokes the `__abs__` magic-method.
+    #[inline]
+    fn absolute(&self, py: Python) -> PyResult<PyObject> {
+        unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_Absolute(self.as_ptr()))
+        }
+    }
+    /// Perform the bitwise negation of self,
+    /// equivalent to the Python expression `~self`
+    ///
+    /// Invokes the `__invert__` magic-method
+    #[inline]
+    fn bitwise_invert(&self, py: Python) -> PyResult<PyObject> {
+        unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_Invert(self.as_ptr()))
+        }
+    }
+    /// Shift this value to the left by the specified number of bits,
+    /// equivalent to the Python expression `self << bits`
+    ///
+    /// Invokes the `__lshift__` magic-method
+    #[inline]
+    fn left_shift(&self, py: Python, bits: impl ToPyObject) -> PyResult<PyObject> {
+        bits.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_Lshift(self.as_ptr(), other))
+        })
+    }
+    /// Shift this value to the right by the specified number of bits,
+    /// equivalent to the Python expression `self >> bits`
+    ///
+    /// Invokes the `__rshift__` magic-method
+    #[inline]
+    fn right_shift(&self, py: Python, bits: impl ToPyObject) -> PyResult<PyObject> {
+        bits.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_Rshift(self.as_ptr(), other))
+        })
+    }
+    /// Perform the "bitwise and" of `self & other`
+    ///
+    /// Invokes the `__and__` magic-method.
+    #[inline]
+    fn bitwise_and(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {
+        other.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_And(self.as_ptr(), other))
+        })
+    }
+    /// Perform the "bitwise exclusive or",
+    /// equivalent to Python expression `self ^ other`
+    ///
+    /// Invokes the `__xor__` magic-method.
+    #[inline]
+    fn bitwise_xor(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {
+        other.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_Xor(self.as_ptr(), other))
+        })
+    }
+    /// Perform the "bitwise or" of `self | other`
+    ///
+    /// Invokes the `__or__` magic-method.
+    #[inline]
+    fn bitwise_or(&self, py: Python, other: impl ToPyObject) -> PyResult<PyObject> {
+        other.with_borrowed_ptr(py, |other| unsafe {
+            err::result_from_owned_ptr(py, ffi::PyNumber_Or(self.as_ptr(), other))
+        })
+    }
+    /// Convert this object to an integer,
+    /// equivalent to the builtin function `int(self)`
+    ///
+    /// Invokes the `__int__` magic-method.
+    ///
+    /// Throws an exception if unable to perform
+    /// the conversion.
+    #[inline]
+    fn as_int(&self, py: Python) -> PyResult<PyLong> {
```

Comment:
Following the Rust naming conventions, `to_int` would be a more appropriate name -- to me, `as_` implies the function returns some form of reference.